src/main/cpp/audioengine.cpp

```diff
@@ -418,10 +421,10 @@ namespace AudioEngine
 
 #ifdef RECORD_TO_DISK
         // write the output to disk if a recording state is active
-        if ( Sequencer::playing && ( recordOutput || recordFromDevice ))
+        if (( Sequencer::playing && recordOutputToDisk ) || recordInputToDisk )
         {
```

Comment:
Note that recording solely the device's audio input to storage can be done without the Sequencer playing. As before, recording engine output requires the Sequencer to be running (as otherwise audio events aren't queued for synthesis / playback).